R/render.R

```diff
@@ -391,8 +391,10 @@ render <- function(input,
   perf_timer_start("pandoc")
 
   # compile Rmd to tex when we need to generate --bibliography
-  if (grepl('[.](pdf|tex)$', output_file) &&
-      ('--bibliography' %in% output_format$pandoc$args)) {
+  # or when keep_tex is TRUE
```

Comment:
The problem is that the intermediate tex file is not produced unless `--bibliography` is in `output_format$pandoc$args`, which is not always the case. For example, I run into this issue when compiling the sample rmd file provided by RStudio (File/New File/R Markdown.../Document/PDF) with added `keep_tex = TRUE`.

With rmarkdown release 0.8.1 the tex file is produced as expected.